shared/subscribers/service.go

```diff
@@ -17,32 +17,47 @@ import (
 	"time"
 
```

Comment:
better place than the shared package for this stuff?

Reply:
as mentioned earlier: not having the shared package

Reply:
see previous comment

Reply:
as discussed  'We should follow "the go proverbs", specifically: a little bit of copying is better than a a little dependency.'    is fine for trivial bits of code, but this is clearly not that, and is actually a perfect example of why we should not duplicate code willy nilly, it was pure chance that I stumbled across the duplicate code, had I not we would have been in a situation where the mem leak would have been fixed in datanode but not in core.